controllers/database_reconciler_helpers.go

```diff
@@ -95,22 +99,28 @@ func (r *DatabaseReconciler) handleDelete(ctx context.Context, database *ndbv1al
 		log.Info("Checking if database instance exists")
 		allDatabases, err := ndb_api.GetAllDatabases(ctx, ndbClient)
 		if err != nil {
-			log.Error(err, "An error occurred while trying to get all databases")
+			errStatement := "An error occurred while trying to get all databases"
+			log.Error(err, errStatement)
+			r.recorder.Event(database, "Warning", "DatabaseLookupError", errStatement+". "+err.Error())
 			return requeueOnErr(err)
 		}
 		if len(util.Filter(allDatabases, func(d ndb_api.DatabaseResponse) bool { return d.Id == database.Status.Id })) == 0 {
 			// Could not find the database with the given database id => database instance has been deleted
+			r.recorder.Event(database, "Normal", "DatabaseDeprovisioned", "Database has been deprovisioned from NDB.")
 			log.Info("Database instance not found, attempting to remove database server.")
+			r.recorder.Event(database, "Normal", "DatabaseServerDeprovisioningStarted", "Deprovisioning database server from NDB.")
 			databaseServerId := database.Status.DatabaseServerId
 			// Make a dbserver deprovisioning request to NDB only if the serverId is present in status
 			if databaseServerId != "" {
 				_, err := ndb_api.DeprovisionDatabaseServer(ctx, ndbClient, databaseServerId, *ndb_api.GenerateDeprovisionDatabaseServerRequest())
 				if err != nil {
 					log.Error(err, "Deprovisioning database server request failed.", "database server id", databaseServerId)
+					r.recorder.Event(database, "Normal", "DatabaseServerDeprovisioningFailed", err.Error())
```

Comment:
"Normal" -> "Warning"

Reply:
or "Error"?

Reply:
Kubernetes supports only two types: Info and Warning. 

https://kubernetes.io/docs/reference/kubernetes-api/cluster-resources/event-v1/